pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -346,10 +345,36 @@ func (r *OAMApplicationReconciler) updateStatus(ctx context.Context, ac *v1alpha
 		}
 	}
 	ac.Status.Workloads = append(ac.Status.Workloads, revisionStatus...)
-
+	// patch the extra fields in the status that is wiped by the Status() function
+	patchExtraStatusField(&ac.Status, acPatch.Status)
 	ac.SetConditions(v1alpha1.ReconcileSuccess())
 }
 
+func patchExtraStatusField(acStatus *v1alpha2.ApplicationConfigurationStatus, acPatchStatus v1alpha2.ApplicationConfigurationStatus) {
```

Comment:
This function has  a problem, workload status which should be removed will always leave here now.